### DISCUSSION TITLE
Can't get a JSON Custom Scalar working

### DISCUSSION THREAD

**Discussion by ryan-coterie** [ExpediaGroup/graphql-kotlin#1432] [Q&A]
Edited my title and post to be a little more accurate but I can't seem to get a JSON Scalar working for a 3rd party integration. The field is I think just a basic JSON object but when trying to follow the instructions for creating a custom Scalar converter my build functionality completely breaks down with the error shown along with my code below:

My code for the converter looks like:
```
class JSONScalarConverter : ScalarConverter<JsonObject> {
    override fun toJson(value: JsonObject): Any = value
    override fun toScalar(rawValue: Any): JsonObject = mapper.readValue(rawValue.toString(), JsonObject::class.java)
}

```

and my Gradle configuration looks like:

```
val graphqlGenerateAnvilClient by tasks.creating(GraphQLGenerateClientTask::class) {
        packageName.set("com.my.packagename.generated")
        schemaFile.set(file("${project.projectDir}/src/main/resources/gql/my.graphql"))
    queryFileDirectory.set(file("${project.projectDir}/src/main/resources/gql/"))
        customScalars.set(listOf(GraphQLScalar(
                "JSON",
                "com.google.gson.JsonObject",
                "com.my.packagename.util.JSONScalarConverter"
        )))
}
```

But I keep seeing this error:

```
Caused by: java.util.NoSuchElementException: No value present
        at com.expediagroup.graphql.plugin.client.generator.types.GenerateTypeNameKt.generateCustomClassName(generateTypeName.kt:98)
```

**Comment by dariuszkuc** [comment]
Hello 👋 
By any chance could you provide a link to a repository that reproduces this issue? Unsure what would be causing this issue.

**Reply by ryan-coterie** [reply]
Hey I actually figured out it was an issue with one of our query files referencing a no longer existing Type in the schema. I didn't realize it had been removed and the error was so generic I thought it had something to do with my Scalar implementation not the query files.
